src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -354,51 +323,47 @@ private async Task UpdateAdvertisingManifestAsync(WorkloadManifestInfo manifest,
-                // if we don't find the file then use the current feature band
-                var adManifestFeatureBandPath = Path.Combine(GetAdvertisingManifestPath(_sdkFeatureBand, manifestId), "AdvertisedManifestFeatureBand.txt");
-
-                SdkFeatureBand adManifestFeatureBand = _sdkFeatureBand;
-                if (File.Exists(adManifestFeatureBandPath))
-                {
-                    adManifestFeatureBand = new SdkFeatureBand(File.ReadAllText(adManifestFeatureBandPath));
-                }
-
+            using FileStream fsSource = new(manifestPath, FileMode.Open, FileAccess.Read);
+            var manifest = WorkloadManifestReader.ReadWorkloadManifest(manifestId.ToString(), fsSource, manifestPath);
+            // we need to know the feature band of the advertised manifest (read it from the AdvertisedManifestFeatureBand.txt file)
+            // if we don't find the file then use the current feature band
+            var adManifestFeatureBandPath = Path.Combine(GetAdvertisingManifestPath(_sdkFeatureBand, manifestId), "AdvertisedManifestFeatureBand.txt");
 
-                return (new ManifestVersion(manifest.Version), adManifestFeatureBand, manifest.Workloads.Values.OfType<WorkloadDefinition>().ToDictionary(w => w.Id));
+            SdkFeatureBand adManifestFeatureBand = _sdkFeatureBand;
+            if (File.Exists(adManifestFeatureBandPath))
+            {
+                adManifestFeatureBand = new SdkFeatureBand(File.ReadAllText(adManifestFeatureBandPath));
             }
+
+            ManifestVersionWithBand manifestWithBand = new(new ManifestVersion(manifest.Version), adManifestFeatureBand);
+            var workloads = manifest.Workloads.Values.OfType<WorkloadDefinition>().ToDictionary(w => w.Id);
+            return (manifestWithBand, workloads);
         }
 
-        private (ManifestVersion manifestVersion, SdkFeatureBand sdkFeatureBand) GetInstalledManifestVersion(ManifestId manifestId)
+        private ManifestVersionWithBand GetInstalledManifestVersion(ManifestId manifestId)
         {
-
-            var manifest = _workloadResolver.GetInstalledManifests()
-                .FirstOrDefault(manifest => manifest.Id.ToLowerInvariant().Equals(manifestId.ToString()));
+            var manifest = _workloadResolver.GetInstalledManifests().FirstOrDefault(manifest => manifest.Id.ToLowerInvariant().Equals(manifestId.ToString()));
             if (manifest == null)
             {
                 throw new Exception(string.Format(LocalizableStrings.ManifestDoesNotExist, manifestId.ToString()));
             }
-            return (new ManifestVersion(manifest.Version), new SdkFeatureBand(manifest.ManifestFeatureBand));
+            var version = new ManifestVersion(manifest.Version);
+            var band = new SdkFeatureBand(manifest.ManifestFeatureBand);
+            return new(version, band);
```

Comment:
I kinda prefer the collapsed form (just with a 'new' in front)

Reply:
So it would be:
```C#
return new(new ManifestVersion(manifest.Version), new SdkFeatureBand(manifest.ManifestFeatureBand));
```

Or I could also do:
```C#
ManifestVersion version = new(manifest.Version);
SdkFeatureBand band = new(manifest.ManifestFeatureBand);
return new(version, band);
```

Preference? I'm actually not against the second one now that I see it. 🤔

Reply:
I prefer the first. If constructing one or both parts is complicated (like if there have to be nested constructors under it), I think it's good to separate them out...but these are quite simple.

Reply:
I've changed this to the first (single statement return).